src/js/angular/core/services/jwt-auth.service.js

```diff
@@ -280,6 +280,15 @@ angular.module('graphdb.framework.core.services.jwtauth', [
                     $rootScope.deniedPermissions = {};
                     this.securityInitialized = true;
 
+                    const selectedRepo = {
+                        id: LocalStorageAdapter.get(LSKeys.REPOSITORY_ID) || '',
+                        location: LocalStorageAdapter.get(LSKeys.REPOSITORY_LOCATION) || ''
+                    };
+
+                    if (!jwtAuth.canReadRepo(selectedRepo)) {
+                        LocalStorageAdapter.remove(LSKeys.REPOSITORY_ID);
+                    }
+
```

Comment:
We probable need to reset and `$rootScope.deniedPermissions`. Can you check this. You can see comment in `repositories.servcice.js setRepository` function.